### DISCUSSION TITLE
Read JSON Lines format

### DISCUSSION THREAD

**Discussion by tznind** [HicServices/RDMP#560] [Ideas]
**Is your feature request related to a problem? Please describe.**
A DLE attacher that can read `.json` and populate RAW tables is sought.

If we can convert json to `System.DataSet` using a library then this should be very easy

**Comment by jas88** [comment]
Sounds like a job for https://www.nuget.org/packages/System.Text.Json/6.0.0-preview.5.21301.5?

**Comment by tznind** [comment]
> 
> 
> Sounds like a job for https://www.nuget.org/packages/System.Text.Json/6.0.0-preview.5.21301.5?

I think we already have a dependency on `using Newtonsoft.Json;` and it looks like it should be relatively easy to get the data out.

Then we either need to put it into a `System.DataSet` or a collection of `System.DataTable` and use the bulk insert feature of FAnsiSql or go the slower route of calling `DiscoveredTable.Insert(Dictionary<DiscoveredColumn,object> values)`.

Hopefully we don't need to deal with auto increment because that is a no go for RDMP DLE.

Nice intro reading:
https://docs.microsoft.com/en-us/sql/relational-databases/json/json-data-sql-server?view=sql-server-ver15

**Comment by tznind** [comment]
I've converted this to a discussion because it would take a lot of work to make this happen and that would mean dedicated resource being allocated to this feature.
